lib/util/packed_xml.js

```diff
@@ -0,0 +1,173 @@
+'use strict';
+
+var lzmaDecoder = require('./lzma.js'),
+    xmldoc = require('xmldoc'),
+    b64_decode = require('./base64.js').b64_decode;
+
+var Stream = {
+    inStream: function(data) {
+        this.offset = 0;
+        this.data = data;
+        this.readByte = function() {
+            return this.data[this.offset++];
+        };
+        this.readUInt32LE = function() {
+            var res = this.data.readUInt32LE(this.offset);
+            this.offset += 4;
+            return res;
+        }
+        return this;
+    },
+    outStream: function(size) {
+        this.offset = 0;
+        this.data = new Buffer(size);
+        this.writeByte = function(value) {
+            this.data[this.offset++] = value;
+        };
+        return this;
+    }
+};
+
+var Unpacker = function() {
+
+    this.getVersion = function(data, dataLen) {
+
+        var bytes = [0x31, 0x32, 0x33, 0x34, 0x35, 0x36, 0x37, 0x38, 0x39, 0x30];
+        var xorStr = [0x0E8, 0x0D5, 1, 3, 0x0C3, 0x0C1, 0x83, 0x3D, 0x0B7, 0x0F0, 0x41, 5, 7, 0x72, 0x10, 0x0E8];
+
+        var res = {'newFormat': false, 'nVer': -1, bRand: 0};
+
+        if (dataLen < 77) {
+            return res;
+        }
+
+        // the magic happens here...
+        for (var count = 0; count < 10; count++) {
+            if (((data[dataLen - 13 + count]) ^ (data[dataLen - 23 + count])) !== bytes[count]) {
+                return res;
+            }
+        }
+
+        res.newFormat = true;
+        res.bRand = data[dataLen - 3];
+        res.nVer = data[dataLen - 2] ^ data[dataLen - 25];
+
+        return res;
+
+    };
+
+    this.unobfuscate = function(packedXmlData) {
+
+        var dataLen, unpackedSize, fileInfo;
+
+        dataLen = packedXmlData.length;
+

```

Comment:
Please separate unpack (unobfuscation + unlzma) from xml processing and transport-level logic.